tests/unit/lfp/lfp.cpp

```diff
@@ -0,0 +1,104 @@
+    for (int i = 0; i < n; i++) {
+        area += f(a + (i + 0.5) * step) * step;  // sum up each small rectangle
+    }
+    return area;
+}
+
+std::array<double, 3> operator*(const double x, const std::array<double, 3>& y) {
+    return {x * y[0], x * y[1], x * y[2]};
+}
+
+std::array<double, 3> operator+(const std::array<double, 3>& x, const std::array<double, 3>& y) {
+    return {x[0] + y[0], x[1] + y[1], x[2] + y[2]};
+}
+
+BOOST_AUTO_TEST_CASE(LFP_PointSource_LineSource) {
+    using namespace coreneuron;
+#if NRNMPI
+    nrnmpi_init(nullptr, nullptr);
+#endif
+    double segment_length{1.0e-6};
+    double segment_start_val{1.0e-6};
+    std::array<double, 3> segment_start = std::array<double, 3>{0.0, 0.0, segment_start_val};
+    std::array<double, 3> segment_end = segment_start +
+                                        std::array<double, 3>{0.0, 0.0, segment_length};
+    double floor{1.0e-6};
+    double pi{3.141592653589};
+
+    std::array<double, 10> vals;
+    double circling_radius{1.0e-6};
+    std::array<double, 3> segment_middle{0.0, 0.0, 1.5e-6};
+    double medium_resistivity_fac{1.0};
+    for (auto k = 0; k < 10; k++) {
+        std::array<double, 3> approaching_elec =
+            segment_middle + std::array<double, 3>{0.0, 1.0e-5 - k * 1.0e-6, 0.0};
+        std::array<double, 3> circling_elec =
+            segment_middle + std::array<double, 3>{0.0,
+                                                   circling_radius * std::cos(2.0 * pi * k / 10),
+                                                   circling_radius * std::sin(2.0 * pi * k / 10)};
+
+        double analytic_approaching_lfp = line_source_lfp_factor(
+            approaching_elec, segment_start, segment_end, floor, medium_resistivity_fac);
+        double analytic_circling_lfp = line_source_lfp_factor(
+            circling_elec, segment_start, segment_end, floor, medium_resistivity_fac);
+        double numeric_circling_lfp = integral(
+            [&](double x) {
+                return 1.0 /
+                       std::max(floor,
+                                norm<std::array<double, 3>, double>(
+                                    circling_elec + (-x) * segment_start + (x - 1) * segment_end));
+            },
+            0.0,
+            1.0,
+            10000);
+        // TEST of analytic vs numerical integration
+        std::clog << "ANALYTIC line source " << analytic_circling_lfp
+                  << " vs NUMERIC line source LFP " << numeric_circling_lfp << "\n";
+        BOOST_REQUIRE_CLOSE(analytic_circling_lfp, numeric_circling_lfp, 1.0e-6);
+        // TEST of LFP Flooring
+        BOOST_REQUIRE((approaching_elec[1] < 0.866e-6) ? analytic_approaching_lfp == 1.0e6 : true);
+        vals[k] = analytic_circling_lfp;
+    }
+    // TEST of SYMMETRY of LFP FORMULA
+    for (size_t k = 0; k < 5; k++) {
+        BOOST_REQUIRE(std::abs((vals[k] - vals[k + 5]) * 0.5 / (vals[k] + vals[k + 5])) < 1.0e-12);
```

Comment:
this is not a norm and we need to take care of division by 0. I would suggest to write:

```suggestion
        BOOST_REQUIRE(std::abs(vals[k] - vals[k + 5]) < 2.0e-12 * max(abs(vals[k], abs(vals[k+5])));
```

or something similar

Reply:
Done.